File: app/src/main/java/com/coinninja/coinkeeper/di/component/CoinKeeperComponent.java
package com.coinninja.coinkeeper.di.component;

import com.coinninja.coinkeeper.cn.wallet.runner.SaveRecoveryWordsRunner;
import com.coinninja.coinkeeper.service.runner.FullSyncWalletRunner;
import com.coinninja.coinkeeper.util.analytics.Analytics;

import java.util.Locale;

public interface CoinKeeperComponent {
    FullSyncWalletRunner getFullSyncRunner();

    SaveRecoveryWordsRunner getSaveRecoveryWordsRunner();

    Analytics getAnalytics();

    Locale getLocale();
}

File: app/src/main/java/com/coinninja/coinkeeper/di/component/CoinKeeperComponent.kt
package com.coinninja.coinkeeper.di.component

import com.coinninja.coinkeeper.twitter.MyTwitterProfile
import com.coinninja.coinkeeper.util.analytics.Analytics
import com.coinninja.coinkeeper.util.image.CircleTransform
import java.util.*

interface CoinKeeperComponent {
    val analytics: Analytics

    val locale: Locale

    fun provideMyTwitter(): MyTwitterProfile
    fun provideCircleTransform(): CircleTransform
}
